src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java

```diff
@@ -0,0 +1,124 @@
+   String currentSourceTable = sourceTable;
+      List<Object> bigQueryRows = new ArrayList<>();
+      getBigQueryTableData(currentSourceTable, bigQueryRows);
+      List<JsonObject> bigQueryResponse = new ArrayList<>();
+      for (Object row : bigQueryRows) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse.add(jsonData);
+      }
+      List<Object> bigQueryRows2 = new ArrayList<>();
+      getBigQueryTableData(currentTargetTable, bigQueryRows2);
+      List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+      for (Object row : bigQueryRows2) {
+        JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+        bigQueryResponse2.add(jsonData);
+      }
+      boolean isValid = compareBigQueryDataAndBQMT(bigQueryResponse, bigQueryResponse2);
+      if (!isValid) {
+        return false; // Return false if validation fails for any table
+      }
+    return true; // Return true if validation passes for all tables
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows) throws IOException,
+    InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static List<String> getTableByName() throws IOException, InterruptedException {
+    List<String> tableNames = new ArrayList<>();
+    List<String> targetTableNames = Arrays.asList("tabA", "tabB");
+    TableResult tableResult = getTableNamesFromDataSet();
+    Iterable<FieldValueList> rows = tableResult.iterateAll();
+
+    for (FieldValueList row : rows) {
+      FieldValue fieldValue = row.get(0);
+      String currentTableName = fieldValue.getStringValue();
+
+      if (targetTableNames.contains(currentTableName)) {
+        tableNames.add(currentTableName);
+      }
+    }
+    if (tableNames.isEmpty()) {
+      throw new IllegalStateException("Tables not found."); // Throw an exception if no tables are found
+    }
+    return tableNames;
+  }
+  private static boolean compareBigQueryDataAndBQMT(List<JsonObject> bigQueryResponse, List<JsonObject> bqmtData) throws NullPointerException {
+    if (bigQueryResponse.size() != bqmtData.size()) {
```

Comment:
NPE is an unchecked exception !
Is this required ?

Reply:
yes this is required